jaxrs20/src/main/java/org/ccci/gto/globalreg/jaxrs20/Jaxrs20GlobalRegistryClient.java

```diff
@@ -0,0 +1,105 @@
+package org.ccci.gto.globalreg.jaxrs20;
+
+import org.ccci.gto.globalreg.BaseGlobalRegistryClient;
+import org.ccci.gto.globalreg.UnauthorizedException;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import java.util.Map;
+
+/**
+ * This class uses the JAX-RS 2 API to interact with the Global Registry.
+ *
+ * The caller may choose the jax-rs implementation by using the {@code Client} or {@code ClientBuilder} constructors.
+ *
+ */
+public class Jaxrs20GlobalRegistryClient extends BaseGlobalRegistryClient
+{
+
+	private final Client client;
+	private final boolean responsibleForClosingClient;
+
+	/**
+	 * Constructs a GR client using {@link ClientBuilder#newBuilder()}.
+	 */
+	public Jaxrs20GlobalRegistryClient() {
+		this(ClientBuilder.newBuilder());
+	}
+
+	/**
+	 * Constructs a GR client that uses the given jax-rs client builder.
+	 * The jax-rs client will be closed when {@link #close()} is invoked.
+	 */
+	public Jaxrs20GlobalRegistryClient(final ClientBuilder clientBuilder) {
+		this(clientBuilder.build(), true);
+	}
+
+	/**
+	 * Constructs a GR client that uses the given jax-rs client.
+	 * The given client will not be closed when {@link #close()} is invoked;
+	 * it is the caller's responsibility.
+	 */
+	public Jaxrs20GlobalRegistryClient(final Client client) {
+		this(client, false);
+	}
+
+	private Jaxrs20GlobalRegistryClient(final Client client, final boolean responsibleForClosingClient) {
+		this.client = client;
+		this.responsibleForClosingClient = false;
```

Comment:
Should set this value to the parameter, not just false.

Reply:
Wah!